amads/time/meter/__init__.py

```diff
@@ -0,0 +1,619 @@
+    granular_pulse: float,
+) -> list:
+    """
+    Create a flat list in the form [4, 1, 2, 1, 3, 1, 2, 1]
+    from the number of intersecting pulses at each position.
+
+    Parameters
+    --------
+    m
+        A level hierarchy, in the form of a list of list,
+        or a metrical hierarchy storing the same.
+    granular_pulse
+        The pulse value of the fastest level to consider e.g., 1, or 0.25.
+
+    Examples
+    --------
+
+    >>> m = [[0.0, 4.0], [0.0, 2.0, 4.0], [0.0, 1.0, 2.0, 3.0, 4.0]]
+    >>> coincident_pulse_list(m, granular_pulse=1)
+    [3, 1, 2, 1]
+
+    You can currently set the `granular_pulse` value to anything (this behavious may change).
+    For instance, a faster level that's not present simply pads the data out:
+
+    >>> m = [[0.0, 4.0], [0.0, 2.0, 4.0], [0.0, 1.0, 2.0, 3.0, 4.0]]
+    >>> coincident_pulse_list(m, granular_pulse=0.5)
+    [3, 0, 1, 0, 2, 0, 1, 0]
+
+    """
+    if isinstance(m, MetricalHierarchy):
+        m = m.start_hierarchy
+    elif not isinstance(m, list):
+        raise ValueError("The X must be a `MetricalHierarchy` object or a list")
+
+    measure_length = m[0][-1]
+
+    for level in m:
+        assert level[-1] == measure_length
+
+    steps = int(measure_length / granular_pulse)
+    granular_level = [granular_pulse * count for count in range(steps)]
+
+    def count_instances(nested_list, target):
+        return sum([sublist.count(target) for sublist in nested_list])
+
+    coincidences = []
+    for target in granular_level:
+        coincidences.append(count_instances(m, target))
+
+    return coincidences
+
+
+# ------------------------------------------------------------------------------
+
+# Subsidiary one-level converters
+
+
+def starts_from_lengths(
+    measure_length: float, pulse_length: float, include_measure_length: bool = True
+):
+    """
+    Convert a pulse length and measure length into a list of starts.
+    All expressed in quarter length.
+
+    Note:
+    A maximum of 4 decimal places is hardcoded.
+    This is to avoid floating point errors or the need for one line of numpy (np.arange)
+    in a module that doesn't otherwise use it.
+    4dp should be sufficient for all realistic use cases.
+
+    Parameters
+    --------
+    measure_length
+        The quarter length of the metrical span overall.
+    pulse_length
+        The quarter length of the pulse (note: must be shorter than the `measure_length`).
+    include_measure_length
+        If True (default) then each level ends with the full cycle length
+        (i.e., the start of the start of the next cycle).
+
+    Examples
+    --------
+
+    >>> starts_from_lengths(4, 1)
+    [0.0, 1.0, 2.0, 3.0, 4.0]
+
+    >>> starts_from_lengths(4, 1, include_measure_length=False)
+    [0.0, 1.0, 2.0, 3.0]
+    """
+    starts = []
+    count = 0
+    while count < measure_length:
+        starts.append(round(float(count), 4))
+        count += pulse_length
+
+    if include_measure_length:
```

Comment:
Maybe a more elegant version of this would be

```
    starts = [round(float(i), 4) for i in range(0, measure_length, pulse_length)]
    if include_measure_length:
        starts.append(round(float(measure_length), 4))
    return starts
```

Reply:
This problem with this is that neither `measure_length` nor `pulse_length` are guaranteed to be integers.

Reply:
Adding an example to that effect.

Reply:
OK, good point, but easily solved by using `numpy.arange` instead of the built-in `range`?